### PR TITLE
- Add french translation.

### PR DESCRIPTION
**Pull request by vlevigneron**
fixes issue zonemaster/zonemaster-engine#808

### CODE REVIEW

**Review by mattias-p — COMMENTED** [review]
I've skimmed the PR and these are the problems I found.

File: share/fr.po [hunk 1]
```diff
@@ -59,6 +59,32 @@ msgstr ""
...
+#. CONNECTIVITY:TEST_CASE_END
+#. CONSISTENCY:TEST_CASE_END
+#. DELEGATION:TEST_CASE_END
+#. DNSSEC:TEST_CASE_END
+#. NAMESERVER:TEST_CASE_END
+#. SYNTAX:TEST_CASE_END
+#. ZONE:TEST_CASE_END
+#, perl-brace-format
+msgid "TEST_CASE_END {testcase}."
+msgstr ""
```
[the remark this answers is not in the document]
Comment (vlevigneron) [reply share/fr.po:73]: Done !

File: share/fr.po [hunk 2]
```diff
@@ -478,7 +507,7 @@ msgstr ""
 "abandonnés."
 
 #. DNSSEC:ALGORITHM_DEPRECATED
-#, perl-brace-format
+#, perl-brace-forma
```
[the remark this answers is not in the document]
Comment (vlevigneron) [reply share/fr.po:510]: Done.

File: share/fr.po [hunk 3]
```diff
@@ -600,6 +648,28 @@ msgstr ""
...
+
+#. DNSSEC:DNSKEY_NOT_ZONE_SIGN
+#, perl-brace-format
+msgid ""
+"Flags field of DNSKEY record with tag {keytag} returned by nameserver {ns} "
+"has not ZONE bit set although DS with same tag is present in parent."
+msgstr ""
+"Le champ \"flags\" de l'enregistrement de type \"DNSKEY\" avec le tag "
+"{keytag} retourné par le serveur de noms {ns} n'a pas le bit ZONE positionné "
+"alors qu'unenregistrement de type \"DS\" avec le même tag de clé existe dans "
```
Comment (pnax) [inline share/fr.po:670]:
```suggestion
"alors qu'un enregistrement de type \"DS\" avec le même tag de clé existe dans "
```

File: share/fr.po [hunk 4]
```diff
@@ -1350,6 +1399,13 @@ msgstr ""
 "Les serveurs de noms de la zone parente ne retournent pas de serveurs ayant "
 "une adresse IPv6. Si il y en avait, il en faudrait au minimum {minimum}."
 
+#. DELEGATION:NS_IS_CNAME
+#, perl-brace-format
+msgid "Nameserver {nsname} RR point to CNAME."
```
Comment (pnax) [inline share/fr.po:1404]:
```suggestion
msgid "Nameserver {nsname} RR points to CNAME."
```

File: share/fr.po [hunk 5]
```diff
@@ -1350,6 +1399,13 @@ msgstr ""
...
+msgstr ""
+"L'enregistrement pour le serveur de noms {nsname} pointe sur un alias "
+"(CNAME)."
```
Comment (pnax) [inline share/fr.po:1407]: Here the French version uses parenthesis around `CNAME` but they are not present in the English version. Is this intended ?

File: share/fr.po [hunk 6]
```diff
@@ -177,59 +203,67 @@ msgstr ""
 "IPv6 est activé, il est possible de réaliser une requête de type "
 "\"{rrtype}\" sur le serveur {ns}."
 
-#. CONNECTIVITY:NAMESERVERS_IPV4_WITH_UNIQ_AS
+#. CONNECTIVITY:ERROR_ASN_DATABASE
 #, perl-brace-format
-msgid ""
-"All nameservers in the delegation have IPv4 addresses in the same AS ({asn})."
+msgid "ASN Database error. No data to analyze for {ns_ip}."
 msgstr ""
```
Comment (pnax) [inline share/fr.po:209]: missing translation, is this intentional ?

File: share/fr.po [hunk 7]
```diff
@@ -177,59 +203,67 @@ msgstr ""
...
-"Toutes les adresses IPv4 des serveurs de noms retournées par les serveurs de "
-"noms de la zone parente se trouvent dans le même AS ({asn})."
 
-#. CONNECTIVITY:NAMESERVERS_IPV6_WITH_UNIQ_AS
+#. CONNECTIVITY:EMPTY_ASN_SET
 #, perl-brace-format
-msgid ""
-"All nameservers in the delegation have IPv6 addresses in the same AS ({asn})."
+msgid "AS database returned no informations for IP address {ns_ip}."
 msgstr ""
```
Comment (pnax) [inline share/fr.po:214]: same, translation is missing here.

File: share/fr.po (hunk 5 above)
Comment (matsduf) [reply share/fr.po:1407]: Well, it says "pointe sur un alias" and then explain that by alias we mean CNAME. I think it looks fine. It is a free translation, and probably a more understandable text than the English version for people with weaker DNS competence.

File: share/fr.po (hunk 6 above)
Comment (matsduf) [reply share/fr.po:209]: There should be a translation.

File: share/fr.po (hunk 7 above)
Comment (matsduf) [reply share/fr.po:214]: Same, same.

File: share/fr.po (hunk 3 above)
Comment (vlevigneron) [reply share/fr.po:670]: Typo fixed

File: share/fr.po (hunk 4 above)
Comment (vlevigneron) [reply share/fr.po:1404]: I will not fix that in this PR. Modification of msgid implies modifications in all langages. Can you create an issue for that ?

File: share/fr.po (hunk 5 above)
Comment (vlevigneron) [reply share/fr.po:1407]: That's the purpose. I guess we can keep it like that.

File: share/fr.po (hunk 6 above)
Comment (vlevigneron) [reply share/fr.po:209]: Fixed

File: share/fr.po (hunk 7 above)
Comment (vlevigneron) [reply share/fr.po:216]: Fixed

**Review by sandoche2k — APPROVED** [review]

File: share/fr.po (hunk 4 above)
Comment (matsduf) [reply share/fr.po:1408]: I created issue #820 for next release (v2020.2).

File: share/fr.po (hunk 5 above)
Comment (matsduf) [reply share/fr.po:1411]: Yes.

**Review by matsduf — COMMENTED** [review]
Looks fine!

**Pull request merged**